model/src/test/java/jetbrains/jetpad/model/transform/HighestPriorityErrorCasesTest.java

```diff
@@ -16,33 +16,60 @@
 package jetbrains.jetpad.model.transform;
 
 import com.google.common.base.Function;
+import com.google.common.base.Functions;
 import jetbrains.jetpad.model.collections.ObservableCollection;
 import jetbrains.jetpad.model.collections.list.ObservableArrayList;
 import org.junit.Before;
 import org.junit.Test;
 
 import java.util.Arrays;
 
-public class HighestPriorityIncorrectPriorityTest {
-  private static final Function<String, Integer> INCONSISTENT_PRIORITY = new Function<String, Integer>() {
+public class HighestPriorityErrorCasesTest {
+  private static final Function<Object, Integer> INCONSISTENT_PRIORITY = new Function<Object, Integer>() {
     private int counter = 0;
     @Override
-    public Integer apply(String s) {
+    public Integer apply(Object s) {
       return counter++;
     }
   };
 
-  private ObservableCollection<String> from;
+  private static final Function<Object, Integer> NULL_PRIORITY = new Function<Object, Integer>() {
+    @Override
+    public Integer apply(Object s) {
+      return null;
+    }
+  };
+
+  private ObservableCollection<Object> from;
 
   @Before
   public void init() {
     from = new ObservableArrayList<>();
-    from.addAll(Arrays.asList("a", "b"));
-    Transformers.highestPriority(INCONSISTENT_PRIORITY).transform(from);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void nullItem() {
+    Transformers.highestPriority(Functions.constant(0)).transform(from);
+    from.add(null);
   }
 
   @Test(expected = IllegalStateException.class)
-  public void remove() {
-    from.remove("a");
+  public void inconsistentPriority() {
+    Transformers.highestPriority(INCONSISTENT_PRIORITY).transform(from);
+    Object sample;
+    from.addAll(Arrays.asList(sample = new Object(), new Object()));
+    from.remove(sample);
+  }
+
+  @Test(expected = NullPointerException.class)
```

Comment:
It is better to throw IllegalState or IllegalArgument exceptions with meaningful clarification messages instead of NPE.